packages/sitecore-jss-react/src/components/EditingScripts.tsx

```diff
@@ -11,24 +12,38 @@ export const EditingScripts = (): JSX.Element => {
     sitecoreContext: { pageState, editMode, clientData, clientScripts },
   } = useSitecoreContext();
 
+  const renderClientData = () => (
+    <>
+      {Object.keys(jssClientData).map((id) => (
+        <script
+          key={id}
+          id={id}
+          type="application/json"
+          dangerouslySetInnerHTML={{ __html: JSON.stringify(jssClientData[id]) }}
+        />
+      ))}
+    </>
+  );
+
   // Don't render anything if not in editing/preview mode
   if (pageState === LayoutServicePageState.Normal) return <></>;
 
+  let jssClientData: Record<string, unknown> = {};
+  if (pageState === LayoutServicePageState.Edit) {
```

Comment:
should this check rather be pageState !== Normal mode to cover other modes?

Reply:
The extra script is for Edit mode only. We only want `jss-hrz-editing` element in Pages edit mode.